meson.build

```diff
@@ -1,6 +1,34 @@
-project('fmt', 'cpp', version : '6.2.0', license: 'BSD', default_options : ['cpp_std=c++14'])
+project('fmt', 'cpp',
+    version : '6.2.0',
+    license : 'BSD',
+    default_options : ['cpp_std=c++14']
+)
 
-inc = include_directories('include')
-lib = library('fmt', ['src/format.cc', 'src/os.cc'], include_directories : inc)
+fmt_private_cpp_args = [ ]
+fmt_interface_cpp_args = [ ]
+libtype = get_option('default_library')
+if libtype == 'shared'
+    fmt_private_cpp_args += [ '-DFMT_EXPORT' ]
+    fmt_interface_cpp_args += [ '-DFMT_SHARED' ]
+endif
 
-fmt_dep = declare_dependency(include_directories : inc, link_with : lib)
+if get_option('fmt_header_only')
+    add_project_arguments('-DFMT_HEADER_ONLY')
```

Comment:
I think you might need to include `-DFMT_HEADER_ONLY` into the `fmt_dep` as well: https://github.com/fmtlib/fmt/issues/596

I think it might be a good idea to 
```
    fmt_private_cpp_args += [ '-DFMT_HEADER_ONLY' ]
    fmt_interface_cpp_args += [ '-DFMT_HEADER_ONLY' ]
```

What do you think?

Reply:
FMT_HEADER_ONLY does not make sense building the library, because there is nothing to build.
FMT_HEADER_ONLY only makes sense for the library user.

I think option which wraps FMT_HEADER_ONLY might add confusion instead of value. We also may need two different deps, because header only dep should not link with the library. I updated the PR in attempt to make it better.

Reply:
Re 1st paragraph: I am sorry if I wasn't clear. The point I was trying to make is `add_project_arguments('-DFMT_HEADER_ONLY')` only adds `FMT_HEADER_ONLY` to the library itself, not the dependencies. So you had to set `fmt_interface_cpp_args` to pass it to users. As you argued `fmt_private_cpp_args` was not necessary.

Re 2nd paragraph: I agree that less options is better. Most users will use the default and I don't test non-default option values on merge. For example your previous code would not work correctly because `language` argument is mandatory: `add_project_arguments('-DFMT_HEADER_ONLY', language: 'cpp')`.

The new style looks much more readable, thank you for improving it.